spec/models/job_post_spec.rb

```diff
@@ -0,0 +1,72 @@
+require 'spec_helper'
+
+describe JobPost do
+  it { should belong_to :school }
+
+  describe "#href" do
+    context "no url provided" do
+      subject { create(:job_post, url: nil) }
+
+      it "is a local link" do
+        expect(subject.href).to
+          eq(Rails.application.routes.url_helpers.job_post_path(subject))
+      end
+    end
+
+    context "url provided" do
+      subject { create(:job_post, url: 'http://stackexchange.com/job/post') }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.